websites/#/라프텔/presence.ts

```diff
@@ -0,0 +1,171 @@
+const presence = new Presence({
+	clientId: "1133347072200949770",
+});
+
+let prevData = "",
+	AnimeData: Detail = {},
+	AnimeDataEpisode: Episode = {};
+
+type Detail = {
+	id?: string;
+	name?: string;
+	img?: string;
+	is_ending?: boolean;
+	animation_info?: {
+		air_year_quarter?: string;
+	};
+	meta_info?: {
+		avg_rating?: string;
+	};
+};
+
+type Episode = {
+	id?: string;
+	title?: string;
+	subject?: string;
+	episode_num?: string;
+};
+
+function getQuery() {
+	return JSON.parse(
+		`{"${decodeURI(location.search.substring(1))
+			.replaceAll('"', '\\"')
+			.replaceAll("&", '","')
+			.replaceAll("=", '":"')}"}`
+	);
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/PDh4ncE.jpg",
+	};
+
+	if (document.location.pathname === "/") {
+		presenceData.smallImageKey = Assets.Search;
+		presenceData.smallImageText = "둘러보는중";
+		presenceData.details = "홈";
+	} else if (document.location.pathname.startsWith("/search")) {
+		presenceData.details = "라프텔 검색";
+		presenceData.smallImageKey = Assets.Search;
+		presenceData.smallImageText = "검색중";
+		presenceData.state = `"${getQuery().keyword}"`;
+	} else if (document.location.pathname.match(/^\/item\/\d/)) {
+		if (prevData === document.location.pathname && AnimeData.name) {
+			presenceData.details = AnimeData.name;
+			presenceData.largeImageKey = AnimeData.img;
+			presenceData.smallImageKey = Assets.VideoCall;
+			presenceData.smallImageText = AnimeData.is_ending ? "완결작품" : "방영중";
+			presenceData.state = AnimeData.animation_info.air_year_quarter;
+
+			presenceData.buttons = [
+				{
+					label: "감상하기",
+					url: `https://laftel.net/item/${AnimeData.id}`,
+				},
+				{
+					label: `별점 ${AnimeData.meta_info.avg_rating}점`,
+					url: `https://laftel.net/item/${AnimeData.id}/review`,
+				},
+			];
+		} else {
+			prevData = document.location.pathname;
+			AnimeData = await (
+				await fetch(
+					`https://laftel.net/api/v1.0/items/${
+						document.location.pathname.split("/")[2]
+					}/detail/`,
+					{
+						headers: {
+							laftel: "TeJava",
+						},
+					}
+				)
+			).json();
+
+			presenceData.details = AnimeData.name;
+			presenceData.largeImageKey = AnimeData.img;
+			presenceData.smallImageKey = Assets.VideoCall;
+			presenceData.smallImageText = AnimeData.is_ending ? "완결작품" : "방영중";
+			presenceData.state = AnimeData.animation_info.air_year_quarter;
+
+			presenceData.buttons = [
+				{
+					label: "감상하기",
+					url: `https://laftel.net/item/${AnimeData.id}`,
+				},
+				{
+					label: `별점 ${AnimeData.meta_info.avg_rating}점`,
+					url: `https://laftel.net/item/${AnimeData.id}/review`,
+				},
+			];
+		}
+	} else if (location.pathname.match(/\/player\/\d*\/\d/)) {
+		const video: HTMLVideoElement = document.querySelector("video");
+		if (video && !isNaN(video.duration)) {
+			if (prevData !== document.location.pathname) {
+				prevData = document.location.pathname;
```

Comment:
Properties of `location` and `document.location` are used many times in this code. To simplify it, destructure them.
```ts
const { pathname, search } = document.location;
```